blog/content/post/2022-04-07-fobnail_attestation_phase.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Fobnail Token - platform attestation
+abstract: 'The Fobnail Token is an open-source hardware USB device that helps to
+           determine the integrity of the system. The purpose of this blog post
+           is to present the development progress of this project. This phase
+           was focused on attestation.'
+cover: /covers/usb_token.png
+author:
+    - artur.kowalski
+    - krystian.hebel
+layout: post
+published: true
+date: 2022-04-06
+archives: "2022"
+
+tags:
+  - fobnail
+  - firmware
+  - tpm
+  - attestation
+categories:
+  - Firmware
+  - Security
+
+---
+
+# About the Fobnail Token project
+
+The Fobnail Token is a project that aims to provide a reference architecture for
+building offline integrity measurement servers on the USB device and clients
+running in Dynamically Launched Measured Environments (DLME). It allows the
+Fobnail owner to verify the trustworthiness of the running system before
+performing any sensitive operation. This project was founded by [NlNet
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/). Also, make sure to
+read other posts related to this project by visiting
+[fobnail](https://www.blog.3mdeb.com/tags/fobnail/) tag.
+
+# Scope of current phase
+
+This phase was mostly about implementing attestation. It includes splitting the
+current implementation of Fobnail (aka Verifier) code into separate modules.
+A module for previous phase (platform provisioning) is started only if Fobnail
+doesn't have platform-specific artifacts saved in flash. A module for
+attestation is always started, either as first stage or immediately after
+provisioning, to test if provisioning succeeded.
+
+We also fixed some hardware issues that forced us to not use nRF52. While it
+still can't be fully used due to inability to provision Fobnail Token itself,
+this moves us closer to the final, hardware, remote (i.e., not running on the
+platform being verified) implementation of Fobnail.
```

Comment:
This is very convoluted paragraph. What we have to say here? Sentences should be simpler.

Reply:
Maybe something as follows, it needs wrapping, and I'm not sure "final hardware, remote implementation of Fobnail" what does it mean.

```suggestion
We also fixed some hardware issues that forced us not to use nRF52. Full utilization is still a problem due to the inability to provision Fobnail Token itself, but released fixes move us closer to the final hardware, remote (i.e., not running on the platform being verified) implementation of Fobnail.
```

Reply:
Maybe because there was no "final hardware", but "final, hardware"? I can change it, but in my opinion this suggestion isn't simpler.

Reply:
@krystian-hebel We need something simpler, I don't get what was the point that why my proposal probably do not address the point.